frontend/test/metabase/scenarios/admin/settings/public-sharing.cy.spec.js

```diff
@@ -0,0 +1,236 @@
+      },
+    })
+      .then(({ body }) => {
+        const questionId = body.id;
+        cy.wrap(questionId).as("questionId");
+        cy.request("POST", `/api/card/${questionId}/public_link`, {});
+      })
+      .then(response => {
+        cy.wrap(response.body.uuid).as("questionUuid");
+      });
+
+    cy.visit("/admin/settings/public-sharing");
+
+    cy.findByText("Shared Questions").should("be.visible");
+    cy.findByText(expectedQuestionName).should("be.visible");
+    cy.get("@questionUuid").then(questionUuid => {
+      cy.findByText(new RegExp(questionUuid)).should("be.visible");
+    });
+
+    cy.get("@questionId").then(questionId => {
+      cy.findByText(expectedQuestionName).should(
+        "have.attr",
+        "href",
+        `/question/${questionId}-${expectedQuestionSlug}`,
+      );
+    });
+
+    cy.button("Revoke link").click();
+    modal().within(() => {
+      cy.findByText("Disable this link?").should("be.visible");
+      cy.button("Yes").click();
+    });
+    cy.findByText("No questions have been publicly shared yet.").should(
+      "be.visible",
+    );
+  });
+
+  it("should see public actions", () => {
+    enableActionsForDB();
+    const expectedActionName = "Public action";
+
+    cy.createQuestion(
+      {
+        name: "Question",
+        query: {
+          "source-table": ORDERS_ID,
+        },
+      },
+      {
+        dataset: true,
+      },
+    ).then(({ body }) => {
+      const modelId = body.id;
+      cy.wrap(modelId).as("modelId");
+    });
+
+    cy.get("@modelId").then(modelId => {
+      createAction({
+        name: expectedActionName,
+        model_id: modelId,
+      }).then(({ body }) => {
+        const actionId = body.id;
+        cy.wrap(actionId).as("actionId");
+      });
+    });
+
+    cy.get("@actionId")
+      .then(actionId => {
+        cy.request("POST", `/api/action/${actionId}/public_link`, {});
+      })
+      .then(({ body }) => {
+        cy.wrap(body.uuid).as("actionUuid");
+      });
+
+    cy.visit("/admin/settings/public-sharing");
+
+    cy.findByText("Shared Actions").should("be.visible");
+    cy.findByText(expectedActionName).should("be.visible");
+    cy.get("@actionUuid").then(actionUuid => {
+      cy.findByText(new RegExp(actionUuid)).should("be.visible");
+    });
+
+    cy.then(function () {
+      cy.findByText(expectedActionName).should(
+        "have.attr",
+        "href",
+        `/model/${this.modelId}/detail/actions/${this.actionId}`,
+      );
+    });
+
+    cy.button("Revoke link").click();
+    modal().within(() => {
+      cy.findByText("Disable this link?").should("be.visible");
+      cy.button("Yes").click();
+    });
+    cy.findByText("No actions have been publicly shared yet.").should(
+      "be.visible",
+    );
+  });
+});
+
+/**
+ *
+ * @param {Partial<import("metabase-types/api").CreateQueryActionApiPayload>} actionDetails
+ */
+function createAction(actionDetails) {
+  const defaultActionDetails = {
```

Comment:
lots of nasty details here, I mostly observe how we're creating a new action and replicate the request here. Not sure if this is the best, but it will do its job for now.

Reply:
I was thinking about that. But, cypress command [is a bit trickier to type](https://docs.cypress.io/guides/tooling/typescript-support#Types-for-Custom-Commands) vs just importing functions that have types of their own, also I [was working on another PR ](https://github.com/metabase/metabase/pull/28215)that would make autoimport possible on Cypress, and I think it'd be more helpful if we create this as a util instead.

The tradeoff is that `createQuestion` and `createQuestionAndAddToDashboard` is a Cypress command, not a function, so we'll have many different ways of dealing with it.

Reply:
I'd follow existing patterns and maybe start a discussion about this. What's the benefit of typing the command if we write E2E tests in JavaScript tho?

Reply:
I might jump the gun a little. I chat with @nemanjaglumac and he asked if we could convert some parts of the Cypress e.g. utils to TypeScript. But I thought about converting the test itself to TS, so just ignore that part about typing custom commands.

The benefit of using functions is that we can try to find the function by having typescript auto-import them vs. having to remember if a certain command exists. To be fair I only know a couple of commands, but there are loads of commands in existence. It should improve these functions' discovery.

To be fair, this sounds like more of a FE discussion thing. I'm fine with sticking to the pattern now (custom command), until we can conclude anything concrete from the sync.

Reply:
I see, makes a ton of sense! I'm personally good with either and would rather support replacing commands with function

Reply:
I leave this as is for now, now I'm out of brain power.